_rules/summary-non-empty-accessible-name-2t702h.md

```diff
@@ -0,0 +1,195 @@
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `summary` element has an [accessible name][] because of its text content.
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 2
+
+This `summary` element has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<details>
+	<summary aria-label="Opening times"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 3
+
+This `summary` element has an [accessible name][] because of its `aria-labelledby` attribute.
+
+```html
+<span id="opening-times">Opening times</span>
+<details>
+	<summary aria-labelledby="opening-times"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 4
+
+This `summary` element has an [accessible name][] because of its text content. It does not need to be the first child element of `details`.
+
+```html
+<details>
+	<p>This is a website. We are available 24/7.</p>
+	<summary>Opening times</summary>
+</details>
+```
+
+#### Passed Example 5
+
+This first `summary` element has an [accessible name][] because of its text content. The second `summary` element is inapplicable because only the first `summary` element will be used as a control for the `details` element.
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `summary` element has no [accessible name][] because it has no content or attribute that can provide it.
+
+```html
+<details>
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Failed Example 2
+
+This `summary` element has an [explicit role][] of `none`. However, it is [focusable][] (by default) which causes [Presentational Roles Conflict Resolution][]. It fails because it has an empty [accessible name][].
+
+```html
+<details>
+	<summary role="none"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+This `summary` element is not a child of a `details` element and so will not be interactive.
+
+```html
+<summary></summary>
+```
+
+#### Inapplicable Example 2
+
+This `summary` element is not a direct child of a `details` element and so will not be interactive.
+
+```html
+<details>
+	<div>
+		<summary></summary>
+	</div>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Inapplicable Example 3
+
+This `summary` element has an explicit semantic role of `button`. These are tested under [Button has non-empty accessible name][97a4e1] instead.
+
+```html
+<details>
+	<summary role="button">Opening hours</summary>
```

Comment:
As per the [ARIA in HTML definition of "No role"](https://www.w3.org/TR/html-aria/#dfn-no-role), I understand that it is an author error to use an explicit role on `summary`. Which is probably not what we want in Inapplicable example (notably, tools checking ARIA  conformance could fail that independently of the rule).

Reply:
I'm not sure I agree with that. Yes this is invalid, but WCAG doesn't require valid markup, and we often have inapplicable examples to demonstrate that invalid markup can still conform WCAG. If someone is failing this as a WCAG issue I think they made a mistake. I think an example that makes doing that an inconsistency with the rule is probably a good thing.

Reply:
As discussed. Added a note on this not being valid ARIA in HTML.